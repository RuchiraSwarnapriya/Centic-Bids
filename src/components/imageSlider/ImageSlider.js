import React from 'react';
import Swiper from 'react-native-swiper';
import { View, StyleSheet, Image } from 'react-native';
import NoPerview from "../../assets/images/no_preview.png";
import { Colors } from "../../assets/colors";


const ImageSlider = ({ images, imageStyle }) => {
    if (images.length == 0) {
        return (
            <View style={styles.wrapper}>
                <View style={styles.slider}>
                    <Image style={[styles.itemImage, { ...imageStyle }]} source={NoPerview} />
                </View>
            </View>
        )
    } else {
        return (
            <Swiper autoplayTimeout={5}
                style={styles.wrapper}
                showsButtons={false}
                loadMinimal={true}
                showsPagination={true}
                paginationStyle={styles.paginationStyle}
                activeDotStyle={styles.activeDotStyle}
                dotStyle={styles.dotStyle}
                loop={true} autoplay={true}
            >
                {images.map((data, index) => {
                    return (
                        <View key={index} style={styles.slider}>
                            <Image style={[styles.itemImage, { ...imageStyle }]} source={{ uri: data }} />
                        </View>
                    )
                })}
            </Swiper>
        )
    }
}

const styles = StyleSheet.create({
    slider: {
        justifyContent: 'center',
        alignItems: 'center',
    },
    activeDotStyle: {
        width: 8,
        height: 8,
        borderRadius: 4,
        backgroundColor: Colors.magenda
    },
    dotStyle: {
        width: 5,
        height: 5,
        borderRadius: 2.5,
        backgroundColor: Colors.darkGrey
    },
    paginationStyle: {
        bottom: -8
    },
    itemImage: {
        alignItems: 'center',
        justifyContent: 'center',
        margin: 15,
        width: 120,
        height: 120,
        borderRadius: 5
    },
});

export default ImageSlider
