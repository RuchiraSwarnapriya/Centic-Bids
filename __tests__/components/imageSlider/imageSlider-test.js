import React from 'react';
import renderer from 'react-test-renderer';
import ImageSlider from '../../../src/components/imageSlider/ImageSlider';


const images = [
]

test('renders correctly', () => {
  const tree = renderer.create(<ImageSlider images={images} />).toJSON();
  expect(tree).toMatchSnapshot();
});