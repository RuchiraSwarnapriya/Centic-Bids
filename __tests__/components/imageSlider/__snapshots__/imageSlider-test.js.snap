// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "position": "relative",
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    activeDotStyle={
      Object {
        "backgroundColor": "#004E94",
        "borderRadius": 4,
        "height": 8,
        "width": 8,
      }
    }
    automaticallyAdjustContentInsets={false}
    autoplay={true}
    autoplayDirection={true}
    autoplayTimeout={5}
    bounces={false}
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    contentOffset={
      Object {
        "x": 0,
      }
    }
    disableNextButton={false}
    disablePrevButton={false}
    dotStyle={
      Object {
        "backgroundColor": "#A5A5A7",
        "borderRadius": 2.5,
        "height": 5,
        "width": 5,
      }
    }
    horizontal={true}
    index={0}
    loadMinimal={true}
    loadMinimalSize={1}
    loop={true}
    onIndexChanged={[Function]}
    onMomentumScrollEnd={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    paginationStyle={
      Object {
        "bottom": -8,
      }
    }
    pagingEnabled={true}
    removeClippedSubviews={true}
    scrollsToTop={false}
    showsButtons={false}
    showsHorizontalScrollIndicator={false}
    showsPagination={true}
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
